rootio/wbuffer.go

```diff
@@ -55,103 +54,108 @@ func NewWBuffer(data []byte, refs map[int64]interface{}, offset uint32, ctx Stre
 	}
 }
 
-func (w *WBuffer) WriteI8(i int8) {
+func (w *WBuffer) WriteI8(v int8) {
 	if w.err != nil {
 		return
 	}
-	v := byte(i)
-	w.w.WriteByte(v)
+	j := byte(v)
+	w.err = w.w.WriteByte(j)
 }
 
-func (w *WBuffer) WriteI16(i int16) {
+func (w *WBuffer) WriteI16(v int16) {
 	if w.err != nil {
 		return
 	}
-	binary.BigEndian.PutUint16(w.w.p, uint16(i))
+	beg := w.w.c
+	end := w.w.c + 2
+	binary.BigEndian.PutUint16(w.w.p[beg:end], uint16(v))
+	w.w.c += 2
 
 }
 
-func (w *WBuffer) WriteI32(i int32) {
+func (w *WBuffer) WriteI32(v int32) {
 	if w.err != nil {
 		return
 	}
-	err := binary.Write(w.w, binary.BigEndian, i)
-	if err != nil {
-		fmt.Println(err)
-	}
+	w.err = binary.Write(w.w, binary.BigEndian, v)
 }
 
-func (w *WBuffer) WriteI64(i int64) {
+func (w *WBuffer) WriteI64(v int64) {
 	if w.err != nil {
 		return
 	}
-	err := binary.Write(w.w, binary.BigEndian, i)
-	if err != nil {
-		fmt.Println(err)
-	}
+	w.err = binary.Write(w.w, binary.BigEndian, v)
 }
 
-func (w *WBuffer) WriteU8(i uint8) {
+func (w *WBuffer) WriteU8(v uint8) {
 	if w.err != nil {
 		return
 	}
-	v := byte(i)
-	w.w.WriteByte(v)
+	j := byte(v)
+	w.err = w.w.WriteByte(j)
 }
 
-func (w *WBuffer) WriteU16(i uint16) {
+func (w *WBuffer) WriteU16(v uint16) {
 	if w.err != nil {
 		return
 	}
-	binary.BigEndian.PutUint16(w.w.p, uint16(i))
+	beg := w.w.c
+	end := w.w.c + 2
+	binary.BigEndian.PutUint16(w.w.p[beg:end], v)
+	w.w.c += 2
 }
 
-func (w *WBuffer) WriteU32(i uint32) {
+func (w *WBuffer) WriteU32(v uint32) {
 	if w.err != nil {
 		return
 	}
-	err := binary.Write(w.w, binary.BigEndian, i)
-	if err != nil {
-		fmt.Println(err)
-	}
+	w.err = binary.Write(w.w, binary.BigEndian, v)
 }
 
-func (w *WBuffer) WriteU64(i uint64) {
+func (w *WBuffer) WriteU64(v uint64) {
 	if w.err != nil {
 		return
 	}
-	err := binary.Write(w.w, binary.BigEndian, i)
-	if err != nil {
-		fmt.Println(err)
-	}
+	w.err = binary.Write(w.w, binary.BigEndian, v)
 }
 
-func (w *WBuffer) WriteF32(i float32) {
+func (w *WBuffer) WriteF32(v float32) {
 	if w.err != nil {
 		return
 	}
-	err := binary.Write(w.w, binary.BigEndian, i)
-	if err != nil {
-		fmt.Println(err)
+	w.err = binary.Write(w.w, binary.BigEndian, v)
+}
+
+func (w *WBuffer) WriteF64(v float64) {
+	if w.err != nil {
+		return
 	}
+	w.err = binary.Write(w.w, binary.BigEndian, v)
 }
 
-func (w *WBuffer) WriteF64(i float64) {
+func (w *WBuffer) WriteBool(v bool) {
 	if w.err != nil {
 		return
 	}
-	err := binary.Write(w.w, binary.BigEndian, i)
-	if err != nil {
-		fmt.Println(err)
+	var o byte = 0
+	if v {
+		o = 1
 	}
+	w.err = binary.Write(w.w, binary.BigEndian, o)
 }
 
-func (w *WBuffer) WriteBool(boolean bool) {
+func (w *WBuffer) WriteString(s string) {
 	if w.err != nil {
 		return
 	}
-	err := binary.Write(w.w, binary.BigEndian, boolean)
-	if err != nil {
-		fmt.Println(err)
+	b1 := []byte(s)
```

Comment:
this is a bit long winded, and full of temporaries.
what about:
```go
b := make([]byte, len(s)+1)
b[0] = len(s)
copy(b[1:], []byte(s)...)
```

also: this will only work "small" strings.